packages/actor-rdf-resolve-hypermedia-qpf/lib/ActorRdfResolveHypermediaQpf.ts

```diff
@@ -0,0 +1,71 @@
+import {AbstractMediatypeUtilities} from "@comunica/actor-abstract-mediatype-utilities";
+import {ISearchForms} from "@comunica/actor-rdf-metadata-extract-hydra-controls";
+import {ActorRdfResolveHypermedia, IActionRdfResolveHypermedia,
+  IActorRdfResolveHypermediaOutput} from "@comunica/bus-rdf-resolve-hypermedia";
+import {ActionContext, IActorArgs, IActorTest} from "@comunica/core";
+
+/**
+ * A comunica QPF RDF Resolve Quad Pattern Actor.
+ */
+export class ActorRdfResolveHypermediaQpf extends ActorRdfResolveHypermedia
+implements IActorRdfResolveHypermediaQpfArgs {
+
+  public readonly subjectUri: string;
+  public readonly predicateUri: string;
+  public readonly objectUri: string;
+  public readonly graphUri?: string;
+
+  constructor(args: IActorRdfResolveHypermediaQpfArgs) {
+    super(args);
+  }
+
+  public async test(action: IActionRdfResolveHypermedia): Promise<IActorTest> {
+    if (!(await AbstractMediatypeUtilities.singleSourceHasType(action.context, 'hypermedia'))) {
+      throw new Error(this.name
+        + ' requires a single source with a Hypermedia \'hypermedia\' entrypoint to be present in the context.');
+    }
+    return true;
+  }
+
+  /**
+   * Look for the search form
+   * @param {IActionRdfResolveHypermedia} the metadata to look for the form.
+   * @return {Promise<IActorRdfResolveHypermediaOutput>} A promise resolving to a hypermedia form.
+   */
+  public async run(action: IActionRdfResolveHypermedia):
+  Promise<IActorRdfResolveHypermediaOutput> {
+
+    if (action.metadata && action.metadata.searchForms) {
```

Comment:
I would move this condition to your test method, as I think you might otherwise run into trouble when you register multiple actors onto this bus.